lms/djangoapps/course_api/tests/test_views.py

```diff
@@ -100,6 +101,15 @@ def test_not_logged_in(self):
         self.client.logout()
         self.verify_response()
 
+    def test_throttle_for_user(self):
+        """Make sure a user requests do not exceed the maximum number of requests"""
```

Comment:
@waheedahmed: You may want to take a look at the test Chris ended up adding for the other throttle change:
https://github.com/edx/edx-platform/pull/18224/files
Chris's test simply makes sure that the throttle was not misconfigured.

This differs from this test, where if we ever have a performant endpoint and bump up the rate, the test will take longer and longer to run and could run into timing issues like some of our older flaky throttling tests.

Reply:
@BenjiLee: I think it would be useful to work out and document some best practices around introducing throttling as you had mentioned.  We should consider testing.  We should note that the method of using switches is available to test different rates if and when needed.

Additionally, @feanil and @BenjiLee, I imagine that when we set up a throttle rate it will be the lower of the following:
1. the limit at which the endpoint can perform, or 
2. the limit at which we think it is reasonable for callers to call the endpoint.
Part of the best practice should probably include which of these we are using.  In other words, if the limit is set to 1 above (performance), and it is being limited by current performance, we should easily know that and know that if we increase performance we still only want to increase the rate as far as 2 above (reasonable usage limit).

Reply:
@BenjiLee want to write an OEP (see above)?  It doesn't need to be extensive, and can just be your quick notes on this topic.  You could timebox it to an hour.

Also, a reminder, we need a CAT-2 ticket for the catalog endpoint rate limiting if we don't already have one.  This too is a CAT-1 waiting to happen.